source/appModules/nvda.py

```diff
@@ -44,6 +54,116 @@ def _get_description(self):
+
+	@script(
+		gesture="kb:alt+upArrow",
+		# Translators: Description of a command to move to the previous result
+		# in the Python Console output pane
+		description=_("Move to the previous result"),
+		category=SCRCAT_PYTHON_CONSOLE
+	)
+	def script_moveToPrevResult(self, gesture: "inputCore.InputGesture"):
+		self._resultNavHelper(direction="previous", select=False)
+
+	@script(
+		gesture="kb:alt+downArrow+shift",
+		# Translators: Description of a command to select from the current caret position to the end
+		# of the current result in the Python Console output pane
+		description=_("Select until the end of the current result"),
+		category=SCRCAT_PYTHON_CONSOLE
+	)
+	def script_selectToResultEnd(self, gesture: "inputCore.InputGesture"):
+		self._resultNavHelper(direction="next", select=True)
+
+	@script(
+		gesture="kb:alt+shift+upArrow",
+		# Translators: Description of a command to select from the current caret position to the start
+		# of the current result in the Python Console output pane
+		description=_("Select until the start of the current result"),
+		category=SCRCAT_PYTHON_CONSOLE
+	)
+	def script_selectToResultStart(self, gesture: "inputCore.InputGesture"):
+		self._resultNavHelper(direction="previous", select=True)
+
+	def _resultNavHelper(self, direction: str = "next", select: bool = False):
+		from pythonConsole import consoleUI
+		startPos, endPos = consoleUI.outputCtrl.GetSelection()
+		if self.isTextSelectionAnchoredAtStart:
+			curPos = endPos
+		else:
+			curPos = startPos
+		if direction == "previous":
+			for pos in reversed(consoleUI.outputPositions):
+				if pos < curPos:
+					break
+			else:
+				# Translators: Reported when attempting to move to the previous result in the Python Console
+				# output pane while there is no previous result.
+				speech.speakMessage(_("Top"))
+				return
+		elif direction == "next":
+			for pos in consoleUI.outputPositions:
+				if pos > curPos:
+					break
+			else:
+				# Translators: Reported when attempting to move to the next result in the Python Console
+				# output pane while there is no next result.
+				speech.speakMessage(_("Bottom"))
+				return
+		else:
+			raise ValueError(u"Unexpected direction: {!r}".format(direction))
+		if select:
+			consoleUI.outputCtrl.Freeze()
+			anchorPos = startPos if self.isTextSelectionAnchoredAtStart else endPos
+			consoleUI.outputCtrl.SetSelection(anchorPos, pos)
+			consoleUI.outputCtrl.Thaw()
+			self.detectPossibleSelectionChange()
+			self.isTextSelectionAnchoredAtStart = anchorPos < pos
+		else:
+			consoleUI.outputCtrl.SetSelection(pos, pos)
+		info = self.makeTextInfo(textInfos.POSITION_CARET)
+		copy = info.copy()
+		info.expand(textInfos.UNIT_LINE)
+		if (
+			copy.move(textInfos.UNIT_CHARACTER, 4, endPoint="end") == 4
+			and copy.text == ">>> "
+		):
+			info.move(textInfos.UNIT_CHARACTER, 4, endPoint="start")
+		speech.speakTextInfo(info, reason=controlTypes.REASON_CARET)

```

Comment:
`controlTypes.REASON_CARET` -> `controlTypes.OutputReason.CARET`

Reply:
Done as of ce92a028c7.
Thanks!